caom2-pkg-server/src/main/java/ca/nrc/cadc/caom2/pkg/CaomPackageRunner.java

```diff
@@ -0,0 +1,269 @@
+    protected String getPackageName() {
+        return this.packageName;
+    }
+
+
+    @Override
+     public Iterator<PackageItem> getItems() throws IOException {
+
+
+        List<PackageItem> packageItems = new ArrayList<PackageItem>();
+        try {
+
+            // obtain credentials from CDP if the user is authorized
+            AccessControlContext accessControlContext = AccessController.getContext();
+            Subject subject = Subject.getSubject(accessControlContext);
+            AuthMethod authMethod = AuthenticationUtil.getAuthMethod(subject);
+            AuthMethod proxyAuthMethod = authMethod;
+
+            if ( CredUtil.checkCredentials() )
+            {
+                proxyAuthMethod = AuthenticationUtil.getAuthMethodFromCredentials(subject);
+            }
+
+            // runID is checked in initPackage()
+            CaomTapQuery query = new CaomTapQuery(tapID, runID);
+
+            CaomArtifactResolver artifactResolver = new CaomArtifactResolver();
+            artifactResolver.setAuthMethod(proxyAuthMethod);
+            artifactResolver.setRunID(runID);
+
+            for (PublisherID publisherID : publisherIDList) {
+                try {
+
+                    // If query fails, the id will be skipped
+                    ArtifactQueryResult result = query.performQuery(publisherID, true);
+
+                    List<Artifact> artifacts = result.getArtifacts();
+                    stripPreviews(artifacts);
+
+                    if (artifacts.isEmpty()) {
+                        // either the input ID was: not found, access-controlled, or has no artifacts
+                        log.info(this.packageName + ": no files available for ID =" + publisherID.getURI().toASCIIString());
+                    } else {
+                        for (Artifact a : artifacts) {
+                            URL url = artifactResolver.getURL(a.getURI());
+
+                            String artifactName = a.getURI().getSchemeSpecificPart();
+                            log.debug("new PackageItem: " + a.getURI() + " from " + url);
+                            log.debug("package entry filename " + artifactName);
+
+                            PackageItem newItem = new PackageItem(url, artifactName);
+                            packageItems.add(newItem);
+                        }
+                    }
+                } catch (ResourceNotFoundException resourceEx) {
+                    // Skip this entry and continue to process
+                    log.info("plane not found: " + publisherID.getURI().toASCIIString() + "skipping...");
+                }
+            }
+        }
+        catch (CertificateException certEx) {
+            // Stop - can be thrown by CredUtil check or TAP query inside for loop
+            log.info("invalid certificate");
+            throw new RuntimeException("invalid certificate", certEx);
+        }
+
+        return packageItems.iterator();
+    }
+
+    // used in an int-test
```

Comment:
package access for test is sufficient (they should be in same package)

also, maybe add a comment here that the assumption is that the publisherID is of the form 
```
ivo://{authority}/{path}/{collection}?{observationID}/{productID}
```
This is true for the java caom2 implementation but isn't strictly required by the model because publisherID is not actually part of the model (reasons). It will be added to the model when I can figure out how to do that and get time to write the paper :-)

Reply:
function access level changed, comment added in javadoc.